### PR TITLE
Document git submodule update in COMPILING.md

### PR DESCRIPTION
**Pull request by johnnonweiler**
Update build documentation (COMPILING.md and jbmc/README.md), given that JBMC now uses a git submodule.  **Also,** this PR has now been extended to remove `--recursive` from the submodule update in the  makefile.

### CODE REVIEW

File: COMPILING.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+# BEFORE COMPILATION
+
+JBMC uses a git submodule. This submodule must be set up before
+attempting a build:
+```
+git submodule update --init --recursive
```
Comment (hannes-steffenhagen-diffblue) [inline COMPILING.md]: The `--recursive` part is strictly speaking not necessary as of now. It doesn't _hurt_ to have in there though, so no real objection.

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

File: COMPILING.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
+# BEFORE COMPILATION
```
Comment (tautschnig) [inline COMPILING.md]: I'd say this should be moved down to the section on jbmc, and the instructions to do `make -C jbmc/src java-models-library-download` should be removed.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 54ae5f84ceb5ae652524da4dbc3afb4e7afed088).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78550218
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Comment by johnnonweiler** [comment]
Following comments from @tautschnig and @peterschrammel, I have now moved the documentation to the sections on jbmc, and changed it to use the `setup-submodules` make target instead of using `git submodule update` directly.

File: COMPILING.md [hunk 3]
```diff
@@ -187,7 +187,7 @@ Follow these instructions:
    ```
 5. To compile JMBC, open the Cygwin shell and type
```
Comment (peterschrammel) [inline COMPILING.md]: JBMC

**Review by peterschrammel — APPROVED** [review]

File: COMPILING.md [hunk 4]
```diff
@@ -240,7 +240,11 @@ require manual modification of build files.
 2. Navigate to the *top level* folder of the project. This is different from
    the Makefile build, which requires you to navigate to the `src` directory
    first.
-3. Generate build files with CMake:
+3. Update git submodules:
+   ```
+   git submodule update --init --recursive
```
Comment (peterschrammel) [inline COMPILING.md]: ` --recursive` is not needed

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: b5075d778491eb1def780eb493b05dda7a39f52d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78562699
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: d5f3c329eb2bcdc74707777e6a5de8cbf9419b38).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78565212
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

File: COMPILING.md (hunk 4 above)
Comment (tautschnig) [reply COMPILING.md]: While I agree, this should really propagate to not use this in `make setup-submodules` either.

**Comment by johnnonweiler** [comment]
At request from @peterschrammel, extra commits have now been added to update jbmc/README.md, and to remove `--recursive` from the submodule update command in the jbmc/src/Makefile.

**Review by tautschnig — APPROVED** [review]

File: jbmc/README.md [hunk 5]
```diff
@@ -27,10 +27,15 @@ JBMC compiles CBMC as part of its build process and as such has all the pre-requ
 Compilation
 ===========
 
-To compile you need to run the command:
+Before compilation, update git submodules using the command:
+
+```bash
+make -C jbmc/src setup-submodules
+```
```
Comment (peterschrammel) [inline jbmc/README.md]: You'll also have to do a `make -C src/minisat2-download` here (which we could avoid if we put our patched version of minisat into a github repo and include it as a submodule as well)

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: f7ddb02b850f683590c86104576007fdae98bb1f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78579201
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

File: jbmc/README.md (hunk 5 above)
Comment (johnnonweiler) [reply jbmc/README.md]: I have added another commit to add this.

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: aad16926be8f9de19dacd175859aa70e4cfdb5f5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78788528
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Pull request merged**